### PR TITLE
Added Highs_setSolution to C API

### LINKED ISSUE #775: setSolution function not available from C API
The function to set the initial solution is not available from the C API.
It would be nice to have it.

### PR DESCRIPTION
**Pull request by jajhall**
but should add testing code to `TestCAPI.c` and for the C++ method before merge. Will close #775

### CODE REVIEW

File: src/interfaces/highs_c_api.h [hunk 1]
```diff
@@ -833,6 +833,24 @@ HighsInt Highs_setBasis(void* highs, const HighsInt* col_status,
...
 HighsInt Highs_setLogicalBasis(void* highs);
 
+/**
+ * Set a solution by passing the column and row primal and dual
+ * solution values. For any values that are unavailable pass NULL.
+ *
+ * @param highs       a pointer to the Highs instance
+ * @param col_value   an array of length [num_col] with the column solution
+ *                    values
+ * @param row_value   an array of length [num_row] with the rowumn solution
```
Comment (odow) [inline src/interfaces/highs_c_api.h]:
```suggestion
 * @param row_value   an array of length [num_row] with the row solution
```

File: src/interfaces/highs_c_api.h [hunk 2]
```diff
@@ -833,6 +833,24 @@ HighsInt Highs_setBasis(void* highs, const HighsInt* col_status,
...
+ *                    values
+ * @param col_dual    an array of length [num_col] with the column dual values
+ * @param row_dual    an array of length [num_row] with the rowumn dual values
```
Comment (odow) [inline src/interfaces/highs_c_api.h]:
```suggestion
 * @param row_dual    an array of length [num_row] with the row dual values
```

**Comment by jajhall** [comment]
Thanks @odow I'd spotted this when editing and then forgot to go back!

**Comment by jajhall** [comment]
This now works, and is tested by hot-starting a MIP in `TestCheckSolution` and `TestCAPI.c`. However, I've written some horrible C to get the full path file name for `check/instances/egout.mps`. Could you tidy this up please, @feldmeier

**Comment by feldmeier** [comment]
Sure - I'll see if I can improve it.

**Comment by jajhall** [comment]
> Sure - I'll see if I can improve it.

Thanks. It's causing some CI failures. Probably trivial for someone like you who understands C

**Pull request merged**